compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json

```diff
@@ -2917,6 +2917,52 @@
                 }
             ]
         },
+        {
+            "name": "MappingBindingPatternNode",
+            "base": "BindingPatternNode",
+            "kind": "MAPPING_BINDING_PATTERN",
+            "attributes": [
+                {
+                    "name": "openBrace",
+                    "type": "Token"
+                },
+                {
+                    "name": "fieldBindingPatterns",
+                    "type": "FieldBindingPatternNode",
+                    "occurrences": "MULTIPLE_SEPARATED"
+                },
+                {
+                    "name": "restBindingPattern",
+                    "type": "RestBindingPatternNode",
+                    "isOptional": true
+                },
+                {
+                    "name": "closeBrace",
+                    "type": "Token"
+                }
+            ]
+        },
+        {
+            "name": "FieldBindingPatternNode",
+            "base": "Node",
+            "kind": "FIELD_BINDING_PATTERN",
+            "attributes": [
+                {
+                    "name": "variableName",
+                    "type": "SimpleNameReferenceNode"
+                },
+                {
+                    "name": "colon",
+                    "type": "Token",
+                    "isOptional": true
+                },
+                {
+                    "name": "bindingPattern",
+                    "type": "BindingPatternNode",
+                    "isOptional": true
+                }
```

Comment:
Since both of these exist together, I think we should have two separate representations for the two kinds of field-binding-patterns